tagstudio/src/qt/widgets/tag_box.py

```diff
@@ -71,36 +71,42 @@ def __init__(self, item, title, field_index, library:Library, tags:list[int], dr
 
 	def set_item(self, item):
 		self.item = item
-
-	def set_tags(self, tags:list[int]):
-		logging.info(f'[TAG BOX WIDGET] SET TAGS: T:{tags} for E:{self.item.id}')
+	
+	def update_tags(self):
 		is_recycled = False
+		
 		if self.base_layout.itemAt(0):
 			# logging.info(type(self.base_layout.itemAt(0).widget()))
+			is_recycled = True
 			while self.base_layout.itemAt(0) and self.base_layout.itemAt(1):
 				# logging.info(f"I'm deleting { self.base_layout.itemAt(0).widget()}")
 				self.base_layout.takeAt(0).widget().deleteLater()
-			is_recycled = True
-		for tag in tags:
+		
+		for tag_id in self.tags:
 			# TODO: Remove space from the special search here (tag_id:x) once that system is finalized.
 			# tw = TagWidget(self.lib, self.lib.get_tag(tag), True, True, 
 			# 							on_remove_callback=lambda checked=False, t=tag: (self.lib.get_entry(self.item.id).remove_tag(self.lib, t, self.field_index), self.updated.emit()), 
 			# 							on_click_callback=lambda checked=False, q=f'tag_id: {tag}': (self.driver.main_window.searchField.setText(q), self.driver.filter_items(q)),
 			# 							on_edit_callback=lambda checked=False, t=tag: (self.edit_tag(t))
 			# 							)
-			tw = TagWidget(self.lib, self.lib.get_tag(tag), True, True)
-			tw.on_click.connect(lambda checked=False, q=f'tag_id: {tag}': (self.driver.main_window.searchField.setText(q), self.driver.filter_items(q)))
-			tw.on_remove.connect(lambda checked=False, t=tag: (self.remove_tag(t)))
-			tw.on_edit.connect(lambda checked=False, t=tag: (self.edit_tag(t)))
+			tw = TagWidget(self.lib, self.lib.get_tag(tag_id), True, True, True)
```

Comment:
Should probably remove the ability to delete "System Tags" (archived & favorite) since TagStudio recreates those tags on system shutdown if they are missing. 
```suggestion
				if tag_id in [0, 1]:  # [0, 1] should probably be extracted as a constant during refactor
					tag_deletable = False
				else:
					tag_deletable = True
				tw = TagWidget(self.lib, self.lib.get_tag(tag_id), True, False, tag_deletable )
```